packages/nextjs/app/builders/0x8D2d6d1A115702FDbFd5704D8b626014E3028C23/page.tsx

```diff
@@ -0,0 +1,203 @@
+  constructor(props: { children: React.ReactNode }) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    console.error("ErrorBoundary caught an error:", error);
+    return { hasError: true, error };
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    console.error("ErrorBoundary caught an error:", { error, errorInfo });
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return (
+        <div className="min-h-screen flex items-center justify-center p-4 bg-base-100">
+          <div className="card bg-error text-error-content">
+            <div className="card-body">
+              <h2 className="card-title">Something went wrong!</h2>
+              <p>Error: {this.state.error?.message}</p>
+              <div className="card-actions">
+                <button
+                  className="btn btn-primary"
+                  onClick={() => this.setState({ hasError: false, error: undefined })}
+                >
+                  Try again
+                </button>
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    }
+
+    return this.props.children;
+  }
+}
+
+// Add error logging with better error details
+if (typeof window !== "undefined") {
+  window.addEventListener("unhandledrejection", event => {
+    console.error("Unhandled promise rejection details:", {
+      reason: event.reason,
+      promise: event.promise,
+      stack: event.reason?.stack,
+      stringified: JSON.stringify(event.reason, null, 2),
+    });
+
+    // Prevent the default behavior of logging "[object Object]"
+    event.preventDefault();
+  });
+
+  window.addEventListener("error", event => {
+    console.error("Global error:", {
+      message: event.message,
+      filename: event.filename,
+      line: event.lineno,
+      error: event.error,
+      stack: event.error?.stack,
+    });
+  });
+}
+// Icon Components - using simple text/emoji approach (no SVG paths!)
+const XIcon = () => <span className="text-lg font-bold">𝕏</span>;
+
+const LinkedInIcon = () => <span className="text-lg font-bold">in</span>;
+
+const GitHubIcon = () => <span className="text-lg">⚡</span>;
+
+const GlobeIcon = () => <span className="text-lg">🌐</span>;
+
+// Social link types
+type IconKey = "x" | "linkedin" | "github" | "globe";
+
+type SocialLink = {
+  href: string;
+  icon: IconKey;
+  title: string;
+};
+
+const socialLinks: SocialLink[] = [
+  { href: "https://x.com/diegobianqui", icon: "x", title: "X (Twitter)" },
+  { href: "https://www.linkedin.com/in/diegobianqui/", icon: "linkedin", title: "LinkedIn" },
+  { href: "https://github.com/diegobianqui", icon: "github", title: "GitHub" },
+  {
+    href: "https://speedrunethereum.com/builders/diegodev.eth",
+    icon: "globe",
+    title: "Speed Run Ethereum",
+  },
+  { href: "https://profiles.cyfrin.io/u/diegobianqui", icon: "globe", title: "Cyfrin Profile" },
+];
```

Comment:
If you use your actual icon directly here instead of a string representation you wouldn't need ifs down on the template. That way, there's only one place to edit when you need more icons.